test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,376 @@
+        cursor_default->set_key(cursor_default.get(), prefix.c_str());
+        cursor_prefix->set_key(cursor_prefix.get(), prefix.c_str());
+
+        int exact_default, exact_prefix;
+        int ret_default = cursor_default->search_near(cursor_default.get(), &exact_default);
+        int ret_prefix = cursor_prefix->search_near(cursor_prefix.get(), &exact_prefix);
+
+        /*
+         * It is not possible to have a prefix search near call successful and the default search
+         * near call unsuccessful.
+         */
+        testutil_assert(
+          ret_default == ret_prefix || (ret_default == 0 && ret_prefix == WT_NOTFOUND));
+
+        /*
+         * We only have to perform validation when the default search near call is successful.
+         */
+        if (ret_default == 0) {
+            /* Both calls are successful. */
+            if (ret_prefix == 0)
+                validate_successful_calls(
+                  cursor_default, cursor_prefix, prefix, exact_default, exact_prefix);
+            /* The prefix search near call failed.*/
+            else
+                validate_unsuccessful_prefix_call(
+                  cursor_default, cursor_prefix, prefix, exact_default, exact_prefix);
+        }
+    }
+
+    void
+    validate_successful_calls(scoped_cursor &cursor_default, scoped_cursor &cursor_prefix,
+      const std::string &prefix, int exact_default, int exact_prefix)
+    {
+        /*
+         * The prefix search near call cannot retrieve a key with a smaller value than the prefix we
+         * searched.
+         */
+        testutil_assert(exact_prefix == 0 || exact_prefix == 1);
+
+        /* Retrieve the keys each cursor is pointing at. */
+        const char *key_default;
+        testutil_check(cursor_default->get_key(cursor_default.get(), &key_default));
+        std::string key_default_str = key_default;
+
+        const char *key_prefix;
+        testutil_check(cursor_prefix->get_key(cursor_prefix.get(), &key_prefix));
+        std::string key_prefix_str = key_prefix;
+
+        logger::log_msg(LOG_TRACE,
+          "search_near (normal) exact " + std::to_string(exact_default) + " key " + key_default);
+        logger::log_msg(LOG_TRACE,
+          "search_near (prefix) exact " + std::to_string(exact_prefix) + " key " + key_prefix);
+
+        /* They key from the prefix search near needs to contain the prefix. */
+        testutil_assert(key_prefix_str.substr(0, prefix.size()) == prefix);
+
+        /*
+         * If the exact value from the default search near call is -1, the key found by the prefix
+         * search near has to be the next key.
+         */
+        if (exact_default == -1) {
+            testutil_check(cursor_default->next(cursor_default.get()));
+            const char *k;
+            testutil_check(cursor_default->get_key(cursor_default.get(), &k));
+            testutil_assert(std::string(k) == key_prefix_str);
```

Comment:
Neat picking: if I'm not mistaken, 'std::string' is redundant at the line above since `std::string` has `operator==` with `const char*`

Reply:
Thanks ! 74311e388952501f114cc5b8815cde12ccf31730